homeassistant/components/slack/notify.py

```diff
@@ -74,16 +82,55 @@
 )
 
 
-async def async_get_service(hass, config, discovery_info=None):
+class AuthDictT(TypedDict, total=False):
+    """Type for auth request data."""
+
+    auth: BasicAuth
+
+
+class FormDataT(TypedDict):
+    """Type for form data, file upload."""
+
+    channels: str
+    filename: str
+    initial_comment: str
+    title: str
+    token: str
+
+
+class MessageT(TypedDict, total=False):
+    """Type for message data."""
+
+    link_names: bool
+    text: str
+    username: str  # Optional
```

Comment:
Since our minimum Python version is 3.7, let's use the standard format:

```python
username: Optionals[str]
icon_url: Optionals[str]
icon_emoji: Optionals[str]
blocks: Optionals[List[Any]]
```

Reply:
The `Optional` comment here is meant to indicate that the key itself is optional in the TypedDict (see `total=False`). I believe one can customize this behavior with inheritance but I haven't really looked into that and it will probably be unnecessary complex.

I'll change the comment to `Optional key` to make it a more clear.